arch/arm/src/nrf52/nrf52_spi.c

```diff
@@ -235,8 +242,12 @@ static struct nrf52_spidev_s g_spi1dev =
   .irq       = NRF52_IRQ_SPI_TWI_1,
 #endif
   .sck_pin   = BOARD_SPI1_SCK_PIN,
+#ifdef BOARD_SPI1_MOSI_PIN
```

Comment:
Is there any point in putting this in the struct a anymore?

We are making our lives hard by having to not encode the information in the pin but then also trying to resolve this at run time. 

The gpio config function could just handle this all on its own checking the pin definition and then we would not have to have these two new functions.